src/cipher.jl

```diff
@@ -110,39 +131,146 @@ function Decryptor(name::AbstractString, key)
     if nettle_major_version >= 3
         ccall( cipher_type.set_decrypt_key, Void, (Ptr{Void}, Ptr{UInt8}), state, pointer(key))
     else
-        ccall( cipher_type.set_decrypt_key, Void, (Ptr{Void}, Cuint, Ptr{UInt8}), state, length(key), pointer(key))
+        ccall( cipher_type.set_decrypt_key, Void, (Ptr{Void}, Cuint, Ptr{UInt8}), state, sizeof(key), pointer(key))
     end
 
     return Decryptor(cipher_type, state)
 end
 
+function decrypt!(state::Decryptor, e::Symbol, iv::Array{UInt8,1}, result, data)
+    if sizeof(result) < sizeof(data)
+        throw(ArgumentError("Output array of length $(sizeof(result)) insufficient for input data length ($(sizeof(data)))"))
+    end
+    if sizeof(result) % state.cipher_type.block_size > 0
+        throw(ArgumentError("Output array of length $(sizeof(result)) must be N times $(state.cipher_type.block_size) bytes long"))
+    end
+    if sizeof(data) % state.cipher_type.block_size > 0
+        throw(ArgumentError("Input array of length $(sizeof(data)) must be N times $(state.cipher_type.block_size) bytes long"))
+    end
+    if sizeof(iv) != state.cipher_type.block_size
+        throw(ArgumentError("Iv must be $(state.cipher_type.block_size) bytes long"))
+    end
+    if ! (symbol(uppercase(string(e))) in _cipher_suites)
+        throw(ArgumentError("now supports $(_cipher_suites) only but ':$(e)'"))
+    end
+if VERSION >= v"0.4.0"
```

Comment:
So does this mean that only CBC mode is supported for versions less than `v0.4.0`?